Backport of Debian patch
* Ignore broadcasted PADO-packets instead of packets from non-unique source
address. Thanks to Stephane Louise <luigi@nerim.net> for detection
and the patch. Closes: #293811

diff -BurpN rp-pppoe-3.10/src/discovery.c rp-pppoe/src/discovery.c
--- rp-pppoe-3.10/src/discovery.c	2008-06-30 14:00:42.000000000 +0000
+++ rp-pppoe/src/discovery.c	2010-12-02 09:31:08.730916002 +0000
@@ -404,8 +404,8 @@ waitForPADO(PPPoEConnection *conn, int t
 	if (!packetIsForMe(conn, &packet)) continue;
 
 	if (packet.code == CODE_PADO) {
-	    if (NOT_UNICAST(packet.ethHdr.h_source)) {
-		printErr("Ignoring PADO packet from non-unicast MAC address");
+	    if (BROADCAST(packet.ethHdr.h_source)) {
+		printErr("Ignoring PADO packet from broadcast MAC address");
 		continue;
 	    }
 	    parsePacket(&packet, parsePADOTags, &pc);
